execution_engine/src/core/runtime/externals.rs

```diff
@@ -668,6 +669,18 @@ where
                 )?;
                 Ok(Some(RuntimeValue::I32(api_error::i32_from(ret))))
             }
+
+            FunctionIndex::Blake2b => {
+                let (in_ptr, in_size, out_ptr, out_size): (u32, u32, u32, u32) = Args::parse(args)?;
+                scoped_instrumenter.add_property("in_size", in_size.to_string());
+                scoped_instrumenter.add_property("out_size", out_size.to_string());
+                let input: Vec<u8> = self.bytes_from_mem(in_ptr, in_size as usize)?;
+                let digest = account::blake2b(&input);
```

Comment:
nit: `out_size` looks like its only used only for instrumenter so instead I'd check if `out_size` is equal to `digest.len()` somewhere before `bytes_from_mem` and return `BufferTooSmall` standard error if it doesnt match, otherwise `0` for success.

Reply:
Yes, will do.  Thanks for reminding me of that.